llvm/tools/objwriter/objwriter.cpp

```diff
@@ -471,13 +505,19 @@ int ObjectWriter::EmitSymbolRef(const char *SymbolName,
     return 4;
   }
   case RelocType::IMAGE_REL_BASED_ARM64_PAGEBASE_REL21: {
+    if (ObjFileInfo->getObjectFileType() == ObjFileInfo->IsMachO) {
+      Kind = MCSymbolRefExpr::VK_GOTPAGE;
```

Comment:
I assume this should be `VK_PAGE` and not `VK_GOTPAGE`. We likely want to generate `ARM64_RELOC_PAGE21` / `ARM64_RELOC_PAGEOFF12` and not a mix of `ARM64_RELOC_GOT_LOAD_PAGE21` 
 and `ARM64_RELOC_PAGEOFF12`.

Reply:
Can't remember why it was `VK_GOTPAGE`, but `VK_PAGE` is also working (current issues are unaffected by this change)
```suggestion
      Kind = MCSymbolRefExpr::VK_PAGE;
```

Reply:
The last rebase got this wrong again... aside from that it looks ok.

Reply:
Which particular test is failing with it? How is this "wrong"?

Reply:
It's not easily visible (at least not on small binaries) but it really should be VK_PAGE to match VK_PAGEOFF in the other relocation.